File: BEE 1040.py
n1, n2, n3, n4 = map(float, input().split(" "))

avg = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / (2 + 3 + 4 + 1)

print("Media: {:.1f}".format(avg))

if (avg >= 7):
    print("Aluno aprovado.")
elif (avg < 5):
    print("Aluno reprovado.")
else:
    print("Aluno em exame.")
    score = float(input())
    print("Nota do exame: {:.1f}".format(score))
    new_avg = (avg + score) / 2
    if (new_avg >= 5):
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print("Media final: {:.1f}".format(new_avg))

File: BEE 1040.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
// var lines = input.split('\n');
var sep = [' ', '\n'];
// console.log(separators.join('|'));
var line = input.split(new RegExp(sep.join('|'), 'g'));

var n1 = parseFloat(line.shift());
var n2 = parseFloat(line.shift());
var n3 = parseFloat(line.shift());
var n4 = parseFloat(line.shift());

var avg = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / (2 + 3 + 4 + 1);

console.log("Media:", avg.toFixed(1));

if (avg >= 7){
    console.log("Aluno aprovado.");
}
else if (avg < 5){
    console.log("Aluno reprovado.");
}
else{
    console.log("Aluno em exame.");
    var score = parseFloat(line.shift());
    console.log("Nota do exame:", score.toFixed(1));
    var new_avg = (avg + score) / 2;
    if (new_avg >= 5){
        console.log("Aluno aprovado.");
    }
    else{
        console.log("Aluno reprovado.");
    }
    console.log("Media final:", new_avg.toFixed(1));
}